test/GasAccounting.t.sol

```diff
@@ -0,0 +1,318 @@
+        (, lastAccessedBlock) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(lastAccessedBlock, uint128(block.number));
+        assertEq(mockGasAccounting.bondedTotalSupply(), bondedTotalSupplyBefore);
+        assertEq(mockGasAccounting.deposits(), depositsBefore);
+
+        bondedTotalSupplyBefore = mockGasAccounting.bondedTotalSupply();
+        depositsBefore = mockGasAccounting.deposits();
+        assertTrue(mockGasAccounting.assign(solverOp.from, assignedAmount));
+        (, lastAccessedBlock) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(lastAccessedBlock, uint128(block.number));
+        assertEq(mockGasAccounting.bondedTotalSupply(), bondedTotalSupplyBefore);
+        assertEq(mockGasAccounting.deposits(), depositsBefore);
+        (, unbonding) = mockGasAccounting.balanceOf(solverOp.from);
+        (bonded,) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(unbonding, 0);
+        assertEq(bonded, 0);
+
+        uint256 unbondingAmount = assignedAmount * 2;
+        mockGasAccounting.increaseUnbondingBalance(solverOp.from, unbondingAmount);
+        bondedTotalSupplyBefore = mockGasAccounting.bondedTotalSupply();
+        depositsBefore = mockGasAccounting.deposits();
+        assertFalse(mockGasAccounting.assign(solverOp.from, assignedAmount));
+        (, lastAccessedBlock) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(lastAccessedBlock, uint128(block.number));
+        assertEq(mockGasAccounting.bondedTotalSupply(), bondedTotalSupplyBefore - assignedAmount);
+        assertEq(mockGasAccounting.deposits(), depositsBefore + assignedAmount);
+        (, unbonding) = mockGasAccounting.balanceOf(solverOp.from);
+        (bonded,) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(unbonding, unbondingAmount + bonded - assignedAmount);
+        assertEq(bonded, 0);
+
+        uint256 bondedAmount = assignedAmount * 3;
+        mockGasAccounting.increaseBondedBalance(solverOp.from, bondedAmount);
+        bondedTotalSupplyBefore = mockGasAccounting.bondedTotalSupply();
+        depositsBefore = mockGasAccounting.deposits();
+        (, uint128 unbondingBefore) = mockGasAccounting.balanceOf(solverOp.from);
+        assertFalse(mockGasAccounting.assign(solverOp.from, assignedAmount));
+        (, lastAccessedBlock) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(lastAccessedBlock, uint128(block.number));
+        assertEq(mockGasAccounting.bondedTotalSupply(), bondedTotalSupplyBefore - assignedAmount);
+        assertEq(mockGasAccounting.deposits(), depositsBefore + assignedAmount);
+        (, unbonding) = mockGasAccounting.balanceOf(solverOp.from);
+        (bonded,) = mockGasAccounting.accessData(solverOp.from);
+        assertEq(unbonding, unbondingBefore);
+        assertEq(bonded, bondedAmount - assignedAmount);
+    }
+
+    function test_credit() public {
+        uint256 creditedAmount = 10_000;
```

Comment:
Looks like it needs > uint128.max boundary tests to me too.

Reply:
agreed

Reply:
Good catch, added